apps/dashboard/src/components/header-navigation/edit-bridge-url-button.tsx

```diff
@@ -0,0 +1,123 @@
+import { useLayoutEffect, useState } from 'react';
+import { RiLinkM, RiPencilFill } from 'react-icons/ri';
+import { PopoverPortal } from '@radix-ui/react-popover';
+import { SubmitHandler, useForm } from 'react-hook-form';
+// eslint-disable-next-line
+// @ts-ignore
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as z from 'zod';
+
+import { cn } from '@/utils/ui';
+import { Popover, PopoverContent, PopoverTrigger } from '../primitives/popover';
+import { Button } from '../primitives/button';
+import { Input, InputField, InputGroup, InputHint, InputLabel } from '../primitives/input';
+import { useBridgeHealthCheck, useUpdateBridgeUrl, useValidateBridgeUrl } from '@/hooks';
+import { ConnectionStatus } from '@/utils/types';
+import { useEnvironment } from '@/context/environment/hooks';
+
+type FormFields = {
+  bridgeUrl: string;
+};
+
+const schema = z.object({ bridgeUrl: z.string().url() });
+
+export const EditBridgeUrlButton = () => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const {
+    register,
+    handleSubmit,
+    reset,
+    setError,
+    formState: { isDirty, errors },
+  } = useForm<FormFields>({ mode: 'onSubmit', resolver: zodResolver(schema) });
+  const { currentEnvironment, setBridgeUrl } = useEnvironment();
+  const { status, bridgeURL: envBridgeUrl } = useBridgeHealthCheck();
+  const { validateBridgeUrl, isLoading: isValidatingBridgeUrl } = useValidateBridgeUrl();
+  const { updateBridgeUrl, isLoading: isUpdatingBridgeUrl } = useUpdateBridgeUrl();
+
+  useLayoutEffect(() => {
+    reset({ bridgeUrl: envBridgeUrl });
+  }, [reset, envBridgeUrl]);
+
+  const onSubmit: SubmitHandler<FormFields> = async ({ bridgeUrl }) => {
+    const { isValid } = await validateBridgeUrl(bridgeUrl);
+    if (isValid) {
+      await updateBridgeUrl({ url: bridgeUrl, environmentId: currentEnvironment?._id ?? '' });
+      setBridgeUrl(bridgeUrl);
+    } else {
+      setError('bridgeUrl', { message: 'The provided URL is not the Novu Endpoint URL' });
+    }
+  };
```

Comment:
When the form is submitted, validate the new Bridge URL, if valid, update it on the environment. Otherwise set the validation error in the form.